user-interface/src/main/java/life/qbic/datamanager/views/general/contact/Contact.java

```diff
@@ -44,7 +42,18 @@ public String getEmail() {
     return email;
   }
 
+  public boolean isEmpty() {
+    return (fullName == null || fullName.isBlank()) && (email == null || email.isBlank());
+  }
+
+  public boolean isComplete() {
+    return fullName != null && !fullName.isBlank() && email != null && !email.isBlank();
+  }
+
   public life.qbic.projectmanagement.domain.model.project.Contact toDomainContact() {
+    if (!isComplete()) {
+      throw new RuntimeException("Contact is not complete and cannot be converted: " + this);
+    }
     return new life.qbic.projectmanagement.domain.model.project.Contact(getFullName(), getEmail());
```

Comment:
Why is there a seperation between contact and domain contact necessary?(What is a domain contact in this context?)
 Was this required by the users and if so do we really want to allow incomplete entries to our database?

Reply:
We do not want to allow incomplete entries, this is why the view bean is checked and only if it is complete, it will be converted to a domain object.
Contact in the domain is a simple value object which has a non-null fullname and an email address.